lib/tariff_jsonapi_parser.rb

```diff
@@ -44,12 +45,17 @@ def parse_collection(collection)
     end
   end
 
-  def parse_top_level_attributes!(resource, parent)
+  def parse_type!(resource, parent)
+    parent.merge!('type' => resource['type'])
```

Comment:
This feels like quite a generic attribute name with potential for clashes with the attributes in the api itself - the original attribute is obviously called `type` but this is internal only and and in JSON-API they hive it off to its own location to avoid clashes, just unfortunate ApiEntity forces us to munge it all together - maybe `resource_type`, `resource_class_name` or `relationship_class` instead - AFAICT its only for our own use internally (currently on the `has_one` in `ApiEntity`) ?

Reply:
I'll use resource_type